src/vs/platform/terminal/electron-browser/localPtyService.ts

```diff
@@ -15,9 +15,17 @@ import { LogLevelChannelClient } from 'vs/platform/log/common/logIpc';
 import { IGetTerminalLayoutInfoArgs, IPtyHostProcessReplayEvent, ISetTerminalLayoutInfoArgs } from 'vs/platform/terminal/common/terminalProcess';
 
 enum Constants {
-	MaxRestarts = 5
+	MaxRestarts = 5,
+	PtyHostIdMask = 0xFFFF0000,
+	RawTerminalIdMask = 0x0000FFFF,
+	PtyHostIdShift = 16
 }
 
+// Tracks the ID of the pty host, when a pty host gets restarted the new one is incremented in order
+// to differentiate the nth terminal from pty host #1 and #2. Since the pty host ID is encoded in
+// the regular ID, we can continue to share IPtyService's interface.
```

Comment:
is there a max number of ptyHosts? 

I find it a bit confusing that a pty host gets a new id when it's restarted.
when pty host 1 is restarted (and thus it's id is now 2), won't it have the same terminals? why are we differentiating between terminals before and after the restart when they should be the same?

Reply:
The pty host could be restarted 5 times if the process dies unexpectedly, but it could theoretically be restarted many times due to unresponsiveness. For you question, when a pty host is restarted, all terminal processes are lost, so pty host id 1+raw terminal id 1 is no longer valid, so we bump the next one to pty host id 2+raw terminal id 1, since only LocalPtyService knows about the pty host id and only PtyService knows about the terminal ID.

We could also think about assigning a pty host ID on launch which is probably the simpler approach.